radiometric_normalization/gimage.py

```diff
@@ -0,0 +1,120 @@
+'''
+Copyright 2015 Planet Labs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+'''
+import logging
+from collections import namedtuple
+
+import numpy
+from osgeo import gdal, gdal_array
+
+# In-memory geospatial image
+# Bands: list of uint16 numpy arrays, each holding a band of data
+# Alpha: uint16 numpy array holding the alpha information
+# Metadata: dict containing georeferencing information
+# (geotransform, projection, rpc)
+GImage = namedtuple('GImage', 'bands, alpha, metadata')
+
+
+def save(gimage, filename):
```

Comment:
It would be nice to expose a save_to_file(gimage, file_like_object) function so that a client can write to in-memory files if they desire. Then this function could be implemented as:

    def save(gimage, filename):
        with open(filename, "w') as file_object:
            save_to_file(gimage, file_object)

This would become important if we wanted to do something like stand up a flask server that generates these types of images on the fly and serves them up to users. If the file gets written to disk, we can serve it no problem but the http request would terminate immediately after serving the file and we'd have no way to clean it up afterwards!

Using a file-like object means we can instead use python's built-in TemoraryFile mechanisms which are destroyed once they go out of scope.